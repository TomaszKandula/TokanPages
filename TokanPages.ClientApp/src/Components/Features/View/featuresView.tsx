import * as React from "react";
import { Link } from "react-router-dom";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import ArrowRightAltIcon from "@material-ui/icons/ArrowRightAlt";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Skeleton from "@material-ui/lab/Skeleton";
import { IGetArticleFeaturesContent } from "../../../Store/States/Content/getArticleFeaturesContentState";
import { RenderCardMedia } from "../../../Shared/Components";
import { ARTICLE_IMAGE_PATH } from "../../../Shared/constants";
import { FeaturesStyle } from "./featuresStyle";
import Validate from "validate.js";

export const FeaturesView = (props: IGetArticleFeaturesContent): JSX.Element =>
{
    const classes = FeaturesStyle();

    const ActiveButton = (): JSX.Element => 
    {
        if (Validate.isEmpty(props.content?.action?.href))
        {
            return (
                <Button endIcon={<ArrowRightAltIcon />} className={classes.button}>
                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.action?.text}
                </Button>
            );
        }

        return(
            <Link to={props.content?.action?.href} className={classes.link}>
                <Button endIcon={<ArrowRightAltIcon />} className={classes.button}>
                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.action?.text}
                </Button>
            </Link>
        );
    }

    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <Box pt={8} pb={12}>
                    <Box textAlign="center" mb={6} data-aos="fade-down">
                        <Typography className={classes.title}>
                            {props.isLoading ? <Skeleton variant="text" /> : props.content?.title.toUpperCase()}
                        </Typography>
                    </Box>
                    <Container maxWidth="lg">
                        <div data-aos="fade-up">
                            <Grid container>
                                <Grid item xs={12} lg={6} className={classes.content}>
                                    <Card elevation={0} className={classes.card}>
                                        <CardContent className={classes.card_content}>
                                            <Box display="flex" flexDirection="column" height="100%" pt={2} px={2}>
                                                <Typography className={classes.text1}>
                                                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.text1}
                                                </Typography>
                                                <Box mt="auto" mb={2}>
                                                    <Typography className={classes.text2}>
                                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.text2}
                                                    </Typography>
                                                </Box>
                                                <Box textAlign="right">
                                                    {props.isLoading ? <Skeleton variant="rect" width="100%" height="25px" /> : <ActiveButton />}
                                                </Box>
                                            </Box>
                                        </CardContent>
                                    </Card>
                                </Grid>
                                <Grid item xs={12} lg={6}>
                                    <Grid container spacing={2}>
                                        <Grid item xs={12} md={8}>
                                            <Card elevation={0} className={classes.card_image}>
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="128px" /> 
                                                    : RenderCardMedia(ARTICLE_IMAGE_PATH, props.content?.image1, classes.media)}
                                            </Card>
                                        </Grid>
                                        <Grid item xs={12} md={4}>
                                            <Card elevation={0} className={classes.card_image}>
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="128px" /> 
                                                    : RenderCardMedia(ARTICLE_IMAGE_PATH, props.content?.image2, classes.media)}
                                            </Card>
                                        </Grid>
                                        <Grid item xs={12} md={4}>
                                            <Card elevation={0} className={classes.card_image}>
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="128px" /> 
                                                    : RenderCardMedia(ARTICLE_IMAGE_PATH, props.content?.image3, classes.media)}
                                            </Card>
                                        </Grid> 
                                        <Grid item xs={12} md={8}>
                                            <Card elevation={0} className={classes.card_image}>
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="128px" /> 
                                                    : RenderCardMedia(ARTICLE_IMAGE_PATH, props.content?.image4, classes.media)}
                                            </Card>
                                        </Grid>
                                    </Grid>
                                </Grid>
                            </Grid>
                        </div>
                    </Container>
                </Box>
            </Container>
        </section>
    );
}
