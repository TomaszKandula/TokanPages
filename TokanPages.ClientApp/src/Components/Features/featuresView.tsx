import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import CodeIcon from "@material-ui/icons/Code";
import LibraryBooksIcon from "@material-ui/icons/LibraryBooks";
import StorageIcon from "@material-ui/icons/Storage";
import CloudIcon from "@material-ui/icons/Cloud";
import Skeleton from "@material-ui/lab/Skeleton";
import ReactHtmlParser from "react-html-parser";
import { IGetFeaturesContent } from "../../Redux/States/getFeaturesContentState";
import featuresStyle from "./featuresStyle";

export default function FeaturesView(props: IGetFeaturesContent)
{
    const classes = featuresStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <div data-aos="fade-up">
                    <Box py={8}>
                        <Box mb={8}>
                            <Typography color="primary" variant="button" component="h3" align="center" gutterBottom={true}>
                                {props.isLoading ? <Skeleton variant="text" /> : props.content?.caption}
                            </Typography>
                            <Typography variant="h4" component="h2" align="center">
                                {props.isLoading ? <Skeleton variant="text" /> : props.content?.header}
                            </Typography>
                        </Box>
                        <Grid container spacing={6}>
                            <Grid item xs={12} sm={6}>
                                <Box mb={2} display="flex" alignItems="center">
                                    <CodeIcon color="primary" className={classes.icon} />
                                    <Typography variant="h5" component="h3">
                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.title1}
                                    </Typography>
                                </Box>
                                <Typography variant="body1" component="p" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : ReactHtmlParser(props.content?.text1)}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6}>
                                <Box mb={2} display="flex" alignItems="center">
                                    <LibraryBooksIcon color="primary" className={classes.icon} />
                                    <Typography variant="h5" component="h3">
                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.title2}
                                    </Typography>
                                </Box>
                                <Typography variant="body1" component="p" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : ReactHtmlParser(props.content?.text2)}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6}>
                                <Box mb={2} display="flex" alignItems="center">
                                    <StorageIcon color="primary" className={classes.icon} />
                                    <Typography variant="h5" component="h3">
                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.title3}
                                    </Typography>
                                </Box>
                                <Typography variant="body1" component="p" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : ReactHtmlParser(props.content?.text3)}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6}>         
                                <Box mb={2} display="flex" alignItems="center">
                                    <CloudIcon color="primary" className={classes.icon} />
                                    <Typography variant="h5" component="h3">
                                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.title4}
                                    </Typography>
                                </Box>
                                <Typography variant="body1" component="p" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : ReactHtmlParser(props.content?.text4)}
                                </Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </div>
            </Container>
        </section>
    );
}