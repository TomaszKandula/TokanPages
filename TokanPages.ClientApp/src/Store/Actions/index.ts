export { ApplicationDialogAction } from "./Application/applicationDialog";
export { ApplicationErrorAction } from "./Application/applicationError";
export { ApplicationLanguageAction } from "./Application/applicationLanguage";
export { ApplicationMessageAction } from "./Application/applicationMessage";

export { ArticleListingAction } from "./Articles/articleListing";
export { ArticleSelectionAction } from "./Articles/articleSelection";
export { ArticleUpdateAction } from "./Articles/articleUpdate";

export { ContentAccountAction } from "./Content/contentAccount";
export { ContentActivateAccountAction } from "./Content/contentActivateAccount";
export { ContentArticleFeaturesAction } from "./Content/contentArticleFeatures";
export { ContentClientsAction } from "./Content/contentClients";
export { ContentContactFormAction } from "./Content/contentContactForm";
export { ContentCookiesPromptAction } from "./Content/contentCookiesPrompt";
export { ContentFeaturedAction } from "./Content/contentFeatured";
export { ContentTechnologiesAction } from "./Content/contentTechnologies";
export { ContentFooterAction } from "./Content/contentFooter";
export { ContentHeaderAction } from "./Content/contentHeader";
export { ContentNavigationAction } from "./Content/contentNavigation";
export { ContentNewsletterAction } from "./Content/contentNewsletter";
export { ContentPolicyAction } from "./Content/contentPolicy";
export { ContentResetPasswordAction } from "./Content/contentResetPassword";
export { ContentStoryAction } from "./Content/contentStory";
export { ContentTermsAction } from "./Content/contentTerms";
export { ContentTestimonialsAction } from "./Content/contentTestimonials";
export { ContentNewsletterRemoveAction } from "./Content/contentNewsletterRemove";
export { ContentUpdatePasswordAction } from "./Content/contentUpdatePassword";
export { ContentNewsletterUpdateAction } from "./Content/contentNewsletterUpdate";
export { ContentUserSigninAction } from "./Content/contentUserSignin";
export { ContentUserSignoutAction } from "./Content/contentUserSignout";
export { ContentUserSignupAction } from "./Content/contentUserSignup";
export { ContentWrongPagePromptAction } from "./Content/contentWrongPagePrompt";

export { NewsletterAddAction } from "./Newsletters/newsletterAdd";
export { NewsletterRemoveAction } from "./Newsletters/newsletterRemove";
export { NewsletterUpdateAction } from "./Newsletters/newsletterUpdate";

export { UserActivateAction } from "./Users/userActivate";
export { UserReAuthenticateAction } from "./Users/userReAuthenticate";
export { UserRemoveAction } from "./Users/userRemove";
export { UserPasswordResetAction } from "./Users/userPasswordReset";
export { UserSigninAction } from "./Users/userSignin";
export { UserSignupAction } from "./Users/userSignup";
export { UserDataStoreAction } from "./Users/userDataStore";
export { UserPasswordUpdateAction } from "./Users/userPasswordUpdate";
export { UserUpdateAction } from "./Users/userUpdate";
export { UserMediaUploadAction } from "./Users/userMediaUpload";
export { UserSignoutAction } from "./Users/userSignout";
