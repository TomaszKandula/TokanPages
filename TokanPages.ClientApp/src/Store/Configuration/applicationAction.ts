import { ApplicationState } from "./applicationState";

export interface ApplicationAction<TAction> {
    (dispatch: (action: TAction) => void, getState: () => ApplicationState): void;
}
