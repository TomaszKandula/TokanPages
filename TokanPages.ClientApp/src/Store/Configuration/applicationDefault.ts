import { ApplicationState } from "./applicationState";
import {
    UserActivate,
    NewsletterAdd,
    ArticleListing,
    ApplicationDialog,
    ApplicationError,
    UserReAuthenticate,
    UserRemove,
    NewsletterRemove,
    UserPasswordReset,
    ArticleSelection,
    ApplicationEmail,
    UserSignin,
    UserSignup,
    UserDataStore,
    ArticleUpdate,
    NewsletterUpdate,
    UserUpdate,
    UserPasswordUpdate,
    ApplicationLanguage,
    UserMediaUpload,
    UserSignout,
    UserNotification,
    UserEmailVerification,
    ApplicationNavbar,
    ContentPageData,
} from "../Defaults";

export const ApplicationDefault: ApplicationState = {
    applicationError: ApplicationError,
    applicationEmail: ApplicationEmail,
    applicationDialog: ApplicationDialog,
    applicationLanguage: ApplicationLanguage,
    applicationNavbar: ApplicationNavbar,
    articleListing: ArticleListing,
    articleSelection: ArticleSelection,
    articleUpdate: ArticleUpdate,
    contentPageData: ContentPageData,
    newsletterAdd: NewsletterAdd,
    newsletterRemove: NewsletterRemove,
    newsletterUpdate: NewsletterUpdate,
    userActivate: UserActivate,
    userEmailVerification: UserEmailVerification,
    userDataStore: UserDataStore,
    userMediaUpload: UserMediaUpload,
    userNotification: UserNotification,
    userPasswordReset: UserPasswordReset,
    userPasswordUpdate: UserPasswordUpdate,
    userReAuthenticate: UserReAuthenticate,
    userRemove: UserRemove,
    userSignin: UserSignin,
    userSignout: UserSignout,
    userSignup: UserSignup,
    userUpdate: UserUpdate,
};
