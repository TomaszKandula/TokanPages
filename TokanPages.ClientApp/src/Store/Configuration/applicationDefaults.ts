import { IApplicationState } from "./applicationState";
import { 
    ActivateAccountDefault,
    AddSubscriberDefault,
    GetAccountContentDefault,
    GetActivateAccountContentDefault,
    GetArticleFeatContentDefault,
    GetClientsContentDefault,
    GetContactFormContentDefault,
    GetCookiesPromptContentDefault,
    GetFeaturedContentDefault,
    GetFeaturesContentDefault,
    GetFooterContentDefault,
    GetHeaderContentDefault,
    GetNavigationContentDefault,
    GetNewsletterContentDefault,
    GetResetPasswordContentDefault,
    GetTestimonialsContentDefault,
    GetUnsubscribeContentDefault,
    GetUpdatePasswordContentDefault,
    GetUpdateSubscriberContentDefault,
    GetUserSigninContentDefault,
    GetUserSignoutContentDefault,
    GetUserSignupContentDefault,
    GetWrongPagePromptContentDefault,
    ListArticlesDefault,
    RaiseDialogDefault,
    RaiseErrorDefault,
    ReAuthenticateUserDefault,
    RemoveAccountDefault,
    RemoveSubscriberDefault,
    ResetUserPasswordDefault,
    SelectArticleDefault,
    SendMessageDefault,
    SigninUserDefault,
    SignupUserDefault,
    StoreUserDataDefault,
    UpdateArticleDefault,
    UpdateSubscriberDefault,
    UpdateUserDefault,
    UpdateUserPasswordDefault,
    UserLanguageDefault,
    GetPolicyContentDefault,
    GetTermsContentDefault,
    GetStoryContentDefault
} from "../Defaults";

export const ApplicationDefaults: IApplicationState = 
{
    applicationDialog: RaiseDialogDefault,
    applicationError: RaiseErrorDefault,
    applicationLanguage: UserLanguageDefault,
    applicationMessage: SendMessageDefault,
    articleListing: ListArticlesDefault,
    articleSelection: SelectArticleDefault,
    articleUpdated: UpdateArticleDefault,
    contentAccount: GetAccountContentDefault,
    contentActivateAccount: GetActivateAccountContentDefault,
    contentArticleFeatures: GetArticleFeatContentDefault,
    contentClients: GetClientsContentDefault,
    contentContactForm: GetContactFormContentDefault,
    contentCookiesPrompt: GetCookiesPromptContentDefault,
    contentFeatured: GetFeaturedContentDefault,
    contentFeatures: GetFeaturesContentDefault,
    contentFooter: GetFooterContentDefault,
    contentHeader: GetHeaderContentDefault,
    contentNavigation: GetNavigationContentDefault,
    contentNewsletter: GetNewsletterContentDefault,
    contentPolicy: GetPolicyContentDefault,
    contentResetPassword: GetResetPasswordContentDefault,
    contentStory: GetStoryContentDefault,
    contentTerms: GetTermsContentDefault,
    contentTestimonials: GetTestimonialsContentDefault,
    contentUnsubscribe: GetUnsubscribeContentDefault,
    contentUpdatePassword: GetUpdatePasswordContentDefault,
    contentUpdateSubscriber: GetUpdateSubscriberContentDefault,
    contentUserSignin: GetUserSigninContentDefault,
    contentUserSignout: GetUserSignoutContentDefault,
    contentUserSignup: GetUserSignupContentDefault,
    contentWrongPagePrompt: GetWrongPagePromptContentDefault,
    subscriberAdd: AddSubscriberDefault,
    subscriberRemove: RemoveSubscriberDefault,
    subscriberUpdate: UpdateSubscriberDefault,
    userActivate: ActivateAccountDefault,
    userDataStore: StoreUserDataDefault,
    userPasswordReset: ResetUserPasswordDefault,
    userPasswordUpdate: UpdateUserPasswordDefault,
    userReAuthenticate: ReAuthenticateUserDefault,
    userRemove: RemoveAccountDefault,
    userSignin: SigninUserDefault,
    userSignup: SignupUserDefault,
    userUpdate: UpdateUserDefault
};
