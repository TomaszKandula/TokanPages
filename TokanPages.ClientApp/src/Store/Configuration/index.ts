export type { IApplicationAction } from "./applicationAction";
export type { IApplicationState } from "./applicationState";

export { ApplicationDefault } from "./applicationDefault";
export { ApplicationReducer } from "./applicationReducer";
export { ConfigureStore } from "./configureStore"; 
