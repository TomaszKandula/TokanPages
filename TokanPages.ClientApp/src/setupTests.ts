// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
