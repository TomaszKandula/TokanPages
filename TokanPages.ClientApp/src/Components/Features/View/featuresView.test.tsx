import "../../../setupTests";
import React from "react";
import * as Redux from "react-redux";
import { shallow } from "enzyme";
import { ApplicationDefault } from "../../../Store/Configuration";
import { FeaturesView } from "./featuresView";

describe("test articles group component: ArticleFeaturesView", () => {
    const testContent = {
        language: "eng",
        title: "Articles",
        description: "I write regularly...",
        text1: ".NET Core, Azure, databases and others.",
        text2: "Let's dive into Microsoft technology...",
        action: {
            text: "View list of articles",
            href: "/action-link",
        },
        image1: "image1.jpg",
        image2: "image2.jpg",
        image3: "image3.jpg",
        image4: "image4.jpg",
    };
    
    let data = ApplicationDefault;
    data.contentArticleFeatures.isLoading = false;
    data.contentArticleFeatures.content = testContent;

    const useSelectorMock = jest.spyOn(Redux, "useSelector");
    const wrapper = shallow(
        <div>
            <FeaturesView />
        </div>
    );

    beforeEach(() => {
        useSelectorMock.mockClear();
        wrapper.find("FeaturesView").dive();
    });
    
    it("should render correctly '<ArticleFeaturesView />' when content is loaded.", () => {
        useSelectorMock.mockReturnValue(data);
        expect(wrapper).toMatchSnapshot();
    });
});
