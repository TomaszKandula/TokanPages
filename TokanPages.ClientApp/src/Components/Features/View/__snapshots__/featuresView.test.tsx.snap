// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test articles group component: ArticleFeaturesView. Renders correctly '<ArticleFeaturesView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <Styled(MuiBox)
      pb={12}
      pt={8}
    >
      <Styled(MuiBox)
        data-aos="fade-down"
        mb={6}
        textAlign="center"
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-title-5"
        >
          ARTICLES
        </WithStyles(ForwardRef(Typography))>
      </Styled(MuiBox)>
      <WithStyles(ForwardRef(Container))
        maxWidth="lg"
      >
        <div
          data-aos="fade-up"
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
          >
            <WithStyles(ForwardRef(Grid))
              className="makeStyles-content-9"
              item={true}
              lg={6}
              xs={12}
            >
              <WithStyles(ForwardRef(Card))
                className="makeStyles-card-2"
                elevation={0}
              >
                <WithStyles(ForwardRef(CardContent))
                  className="makeStyles-card_content-3"
                >
                  <Styled(MuiBox)
                    display="flex"
                    flexDirection="column"
                    height="100%"
                    pt={2}
                    px={2}
                  >
                    <WithStyles(ForwardRef(Typography))
                      className="makeStyles-text1-6"
                    >
                      .NET Core, Azure, databases and others.
                    </WithStyles(ForwardRef(Typography))>
                    <Styled(MuiBox)
                      mb={2}
                      mt="auto"
                    >
                      <WithStyles(ForwardRef(Typography))
                        className="makeStyles-text2-7"
                      >
                        Let's dive into Microsoft technology...
                      </WithStyles(ForwardRef(Typography))>
                    </Styled(MuiBox)>
                    <Styled(MuiBox)
                      textAlign="right"
                    >
                      <ActiveButton />
                    </Styled(MuiBox)>
                  </Styled(MuiBox)>
                </WithStyles(ForwardRef(CardContent))>
              </WithStyles(ForwardRef(Card))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
              lg={6}
              xs={12}
            >
              <WithStyles(ForwardRef(Grid))
                container={true}
                spacing={2}
              >
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  md={8}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Card))
                    className="makeStyles-card_image-4"
                    elevation={0}
                  >
                    <WithStyles(ForwardRef(CardMedia))
                      className="makeStyles-media-8"
                      image="http://localhost:5000/api/v1/assets/getAsset/?BlobName=images/sections/articles/image1.jpg"
                    />
                  </WithStyles(ForwardRef(Card))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  md={4}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Card))
                    className="makeStyles-card_image-4"
                    elevation={0}
                  >
                    <WithStyles(ForwardRef(CardMedia))
                      className="makeStyles-media-8"
                      image="http://localhost:5000/api/v1/assets/getAsset/?BlobName=images/sections/articles/image2.jpg"
                    />
                  </WithStyles(ForwardRef(Card))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  md={4}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Card))
                    className="makeStyles-card_image-4"
                    elevation={0}
                  >
                    <WithStyles(ForwardRef(CardMedia))
                      className="makeStyles-media-8"
                      image="http://localhost:5000/api/v1/assets/getAsset/?BlobName=images/sections/articles/image3.jpg"
                    />
                  </WithStyles(ForwardRef(Card))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  item={true}
                  md={8}
                  xs={12}
                >
                  <WithStyles(ForwardRef(Card))
                    className="makeStyles-card_image-4"
                    elevation={0}
                  >
                    <WithStyles(ForwardRef(CardMedia))
                      className="makeStyles-media-8"
                      image="http://localhost:5000/api/v1/assets/getAsset/?BlobName=images/sections/articles/image4.jpg"
                    />
                  </WithStyles(ForwardRef(Card))>
                </WithStyles(ForwardRef(Grid))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </div>
      </WithStyles(ForwardRef(Container))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
